File: Ejemplo_en_C#/Programa_Video_Juego/Programa_Video_Juego/Clases/Jugador.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Programa_Video_Juego.Clases
{
    class Jugador
    {
        private int vida;
        private List<Arma> l_armas;
        private Arma arma_actual;
        private static int estado_arma = 0;
        private static int tipo_municion = 0;

        public Jugador()
        {
            vida = 30;

            l_armas = new List<Arma>();
            Pistola pistola = new Pistola(6, 2f, 10);
            Escopeta escopeta = new Escopeta(4, 4f, 10);
            Fusil_Asalto fusil = new Fusil_Asalto(15, 3.5f, 10);
            l_armas.Add(pistola);
            l_armas.Add(escopeta);
            l_armas.Add(fusil);

            arma_actual = l_armas[0];

        }

        public int Vida { get => vida; set => vida = value; }
        public List<Arma> L_armas { get => l_armas; set => l_armas = value; }
        public Arma Arma_actual { get => arma_actual; set => arma_actual = value; }

        public string Disparar_Arma(Enemigo objetivo, Video_Juego video_juego)
        {
            
            if (objetivo.Vida > 0)
            {
                video_juego.Puntos += 5;
                return arma_actual.Disparar(objetivo) + "\nSe diparó el arma: " + arma_actual.ToString() + "\nPuntos: " + video_juego.Puntos;
            }
            else if (objetivo.Vida <= 0)
            {
                video_juego.Puntos += 0;
                return arma_actual.Disparar(objetivo) + "\nNo se diparó el arma: " + arma_actual.ToString() + "\nPuntos: " + video_juego.Puntos;
            }
            else
                return "Ocurrio un error";
            
        }

        public string Cambiar_Arma()
        {
            estado_arma++;
            arma_actual = l_armas[estado_arma];
            if (estado_arma == 2)
                estado_arma = -1;
            return "Arma actual: " + arma_actual.ToString();
        }

        public string Recargar_Arma()
        {
            int[] l_municiones = { 6, 4, 15 };
            string[] l_nombres_armas = { "Pistola", "Escopeta", "Fusil de Asalto" };
            if (arma_actual.Municion == 0)
            {
                if (estado_arma == 0) //pistola
                {
                    tipo_municion = 0;
                    arma_actual.Municion += l_municiones[tipo_municion];
                    return "\nSe recargo: " + l_nombres_armas[0] + "\nMunicion: " + arma_actual.Municion;
                }
                else if (estado_arma == 1) //escopeta
                {
                    tipo_municion = 1;
                    arma_actual.Municion += l_municiones[tipo_municion];
                    return "\nSe recargo: " + l_nombres_armas[1] + "\nMunicion: " + arma_actual.Municion;
                }
                else if (estado_arma == 2) //fusil de asalto
                {
                    tipo_municion = 2;
                    arma_actual.Municion += l_municiones[tipo_municion];
                    return "\nSe recargo: " + l_nombres_armas[2] + "\nMunicion: " + arma_actual.Municion;
                }
                else
                    return "\nOcurrio un error";

            }
            else
                return "\nNo se pudo recargar: " + arma_actual.ToString() + "\nMunicion: " + arma_actual.Municion;
        }

    }
}

File: Ejemplo en C/Clases/Jugador.c
#include "string.h"
#include "stdlib.h"

#include "Jugador.h"
#include "Arma.h"
#include "Pistola.h"
#include "Escopeta.h"
#include "Fusil_Asalto.h"

Lista_Armas *Crear_Lista_Armas(Arma *arma_pistola, Arma *arma_escopeta, Arma *arma_fusil_asalto)
{
    Lista_Armas *nueva_lista_armas = (Lista_Armas *)malloc(sizeof(Lista_Armas));

    nueva_lista_armas->arma_escopeta = NULL;
    nueva_lista_armas->arma_fusil_asalto = NULL;
    nueva_lista_armas->arma_pistola = NULL;

    return nueva_lista_armas;
}
void Destruir_Lista_Armas(Lista_Armas *esta_lista_armas)
{
    free(esta_lista_armas);
}

Jugador *Crear_Jugador(int vida)
{
    Jugador *nuevo_jugador = (Jugador *)malloc(sizeof(Jugador));

    Pistola *nueva_pistola = Crear_Pistola(6, 2.3f, 10);
    Escopeta *nueva_escopeta = Crear_Escopeta(4, 4, 20);
    Fusil_Asalto *nuevo_fusil = Crear_Fusil_Asalto(15, 3.5f, 10);

    Lista_Armas *l_armas = Crear_Lista_Armas(nueva_pistola, nueva_escopeta, nuevo_fusil);

    nuevo_jugador->arma_actual = *l_armas->arma_pistola;
    nuevo_jugador->vida = 30;
    nuevo_jugador->estado_arma = 0;
    nuevo_jugador->tipo_municion = 0;

    return nuevo_jugador;
}
void Destruir_Jugador(Jugador *este_jugador)
{
    free(este_jugador);
}

void Disparar_Arma(Enemigo *objetivo, Arma *arma_actual, Video_Juego *video_juego)
{
    char *l_n_armas[] = {"Pistola", "Escopeta", "Fusil de Asalto"};
    if (objetivo->vida > 0 && arma_actual->municion > 0)
    {
        video_juego->puntos += 5;
        Disparar(objetivo, arma_actual);
        printf("Se diparó el arma: %s", ToString(), "\nPuntos: &d", video_juego->puntos);
    }
    else
    {
        printf("Se diparó el arma: %s", ToString(), "\nPuntos: &d", video_juego->puntos);
    }
}
void Cambiar_Arma(Jugador *ref_jugador, Lista_Armas *l_armas)
{
    char *l_n_armas[3] = {"Pistola", "Escopeta", "Fusil de Asalto"};
    ref_jugador->estado_arma++;
    if (ref_jugador->estado_arma == 1)
    {
        ref_jugador->arma_actual = *l_armas->arma_escopeta;
        printf("Arma actual: %s", l_n_armas[1]);
    }
    else if (ref_jugador->estado_arma == 2)
    {
        ref_jugador->arma_actual = *l_armas->arma_fusil_asalto;
        printf("Arma actual: %s", l_n_armas[1]);
    }
    else if (ref_jugador->estado_arma == 3)
    {
        ref_jugador->estado_arma = 0;
        ref_jugador->arma_actual = *l_armas->arma_pistola;
        printf("Arma actual: %s", l_n_armas[1]);
    }
}
void Recargar_Arma(Jugador *ref_jugador, Arma *arma_actual)
{
    int l_municiones[] = {6, 4, 15};
    char *l_n_armas[] = {"Pistola", "Escopeta", "Fusil de Asalto"};
    if (arma_actual->municion == 0)
    {
        if (ref_jugador->estado_arma == 1)
        {
            ref_jugador->tipo_municion = 1;
            ref_jugador->arma_actual.municion += l_municiones[ref_jugador->tipo_municion];
            printf("Se recargo: %s", l_n_armas[1], "\nMunicion: &d", ref_jugador->arma_actual.municion);
        }
        else if (ref_jugador->estado_arma == 2)
        {
            ref_jugador->tipo_municion = 2;
            ref_jugador->arma_actual.municion += l_municiones[ref_jugador->tipo_municion];
            printf("Se recargo: %s", l_n_armas[2], "\nMunicion: &d", ref_jugador->arma_actual.municion);
        }
        else if (ref_jugador->estado_arma == 3)
        {
            ref_jugador->tipo_municion = 0;
            ref_jugador->arma_actual.municion += l_municiones[ref_jugador->tipo_municion];
            printf("Se recargo: %s", l_n_armas[0], "\nMunicion: &d", ref_jugador->arma_actual.municion);
        }
        else
            printf("Ocurrio un error");
    }
}